api/cluster/controller.go

```diff
@@ -190,51 +201,39 @@ func (c *controller) Deploy(ctx context.Context, modelService *models.Service) (
 	}
 
 	isvcName := modelService.Name
-	s, err := c.kserveClient.InferenceServices(modelService.Namespace).Get(isvcName, metav1.GetOptions{})
-	if err != nil {
-		if !kerrors.IsNotFound(err) {
-			log.Errorf("unable to check inference service %s %v", isvcName, err)
-			return nil, ErrUnableToGetInferenceServiceStatus
-		}
-
-		// create new resource
-		spec, err := c.kfServingResourceTemplater.CreateInferenceServiceSpec(modelService, c.deploymentConfig)
-		if err != nil {
-			log.Errorf("unable to create inference service spec %s %v", isvcName, err)
-			return nil, ErrUnableToCreateInferenceService
-		}
 
-		s, err = c.kserveClient.InferenceServices(modelService.Namespace).Create(spec)
-		if err != nil {
-			log.Errorf("unable to create inference service %s %v", isvcName, err)
-			return nil, ErrUnableToCreateInferenceService
-		}
-	} else {
-		// Get current scale of the existing deployment
-		deploymentScale := resource.DeploymentScale{}
+	// Get current scale of the existing deployment
+	deploymentScale := resource.DeploymentScale{}
+	if modelService.RevisionID > 1 {
```

Comment:
`modelService.CurrentIsvcName` should not be empty, because we get the value from `versionEndpoint.InferenceServiceName`

Reply:
Ah, I see what you mean. This comment is related to your other comments ([this](https://github.com/caraml-dev/merlin/pull/471#discussion_r1358062699)). For new endpoint, `versionEndpoint.InferenceServiceName` should be empty, hence `modelService.CurrentIsvcName` also must be empty. Then, if the deployment succeed, we update versionEndpoint.InferenceServiceName which will be saved to DB. Next, if this version endpoint got redeployed, the `versionEndpoint.InferenceServiceName` will already contains the current name (that has revision id suffix). Then we can continue as you suggested. Is my understanding correct?

Reply:
Ok, that is clearer to me now. Although, on this part:

> if this version endpoint got redeployed, the versionEndpoint.InferenceServiceName will already contains the current name

I think it may need to change depending on this discussion: https://github.com/caraml-dev/merlin/pull/471#discussion_r1357985956